hypervisor/arch/x86/timer.c

```diff
@@ -64,40 +66,39 @@ static void local_add_timer(struct per_cpu_timers *cpu_timer,
 
 	list_add(&timer->node, prev);
 
-	if (need_update != NULL) {
-		/* update the physical timer if we're on the timer_list head */
-		*need_update = (prev == &cpu_timer->timer_list);
-	}
+	return (prev == &cpu_timer->timer_list);
 }
 
 int32_t add_timer(struct hv_timer *timer)
 {
 	struct per_cpu_timers *cpu_timer;
 	uint16_t pcpu_id;
-	bool need_update;
+	int32_t ret = 0;
 
 	if ((timer == NULL) || (timer->func == NULL) || (timer->fire_tsc == 0UL)) {
-		return -EINVAL;
+		ret = -EINVAL;
 	}
 
-	ASSERT(list_empty(&timer->node), "add timer again!\n");
+	if (ret == 0) {
```

Comment:
here use "else" is more straight forward and fast.

Reply:
Done